pyuvdata/uvdata/uvdata.py

```diff
@@ -5089,33 +5099,37 @@ def downsample_in_time(
                         )
-                        if summing_correlator_mode:
-                            temp_data[temp_idx] = np.sum(masked_data, axis=0)
-                        else:
-                            # take potential non-uniformity of integration_time into
-                            # account
-                            masked_int_time = np.ma.masked_array(
-                                np.ones_like(
-                                    self.data_array[averaging_idx],
-                                    dtype=self.integration_time.dtype,
-                                )
-                                * self.integration_time[
-                                    averaging_idx, np.newaxis, np.newaxis, np.newaxis
-                                ],
-                                mask=mask,
-                            )
-                            weighted_data = masked_data * masked_int_time
-                            temp_data[temp_idx] = np.sum(
-                                weighted_data, axis=0
-                            ) / np.sum(masked_int_time, axis=0)
+
                         # nsample array is the fraction of data that we actually kept,
                         # relative to the amount that went into the sum or average
                         masked_nsample = np.ma.masked_array(
                             self.nsample_array[averaging_idx], mask=mask
                         )
-                        temp_nsample[temp_idx] = np.sum(masked_nsample, axis=0) / float(
-                            self.flag_array[averaging_idx].shape[0]
+
+                        masked_int_time = np.ma.masked_array(
+                            np.ones_like(
+                                self.data_array[averaging_idx],
+                                dtype=self.integration_time.dtype,
+                            )
+                            * self.integration_time[
+                                averaging_idx, np.newaxis, np.newaxis, np.newaxis
+                            ],
+                            mask=mask,
                         )
+                        if summing_correlator_mode:
+                            temp_data[temp_idx] = np.sum(masked_data, axis=0)
+                        else:
+                            # take potential non-uniformity of integration_time
+                            # and nsamples into account
+                            weights = masked_nsample * masked_int_time
+                            weighted_data = masked_data * weights
+                            temp_data[temp_idx] = np.sum(
+                                weighted_data, axis=0
+                            ) / np.sum(weights, axis=0)
+
+                        temp_nsample[temp_idx] = np.sum(
+                            masked_nsample * masked_int_time, axis=0
+                        ) / np.sum(self.integration_time[averaging_idx])
```

Comment:
Why is the denominator not masked like the numerator?

Reply:
What I was thinking was that we wanted to have the full integration time in the denominator because the nsample is supposed to give you the fraction of the integration that was not flagged before the average was done. But I might not have it right here.

Reply:
yup you're right. I keep forgetting the golden rule: actual integration time = nsample * int time.